### ISSUE TITLE
[Question]: Consistently Hitting org.neo4j.internal.kernel.api.exceptions.schema.IllegalTokenNameException 

### ISSUE THREAD

**Issue by kevin-yang-racap** [run-llama/llama_index#17136]
### Question Validation

- [X] I have searched both the documentation and discord for an answer.

### Question

I am consistently hitting this exception:

`{code: Neo.ClientError.Procedure.ProcedureCallFailed} {message: Failed to invoke procedure `apoc.create.addLabels`: Caused by: org.neo4j.internal.kernel.api.exceptions.schema.IllegalTokenNameException: '' is not a valid token name. Token names cannot be empty or contain any null-bytes.}`

I am trying to build a property graph using the dynamic path llm with Claude Sonnet 3.5 as the LLM. 

Is there some way to fix or avoid this?

**Comment by kevin-yang-racap** [comment]
@dosu, how would I add the function to the preprocessing step? I'm not sure which step to add it in? I'm currently calling from_documents using the extractor.

**Comment by logan-markewich** [comment]
is the llm extracting entities or relations with empty names? I have a feeling claude might be outputting text thats hard to parse

I recommend something like dosu is suggesting. I'd go a step further and even use a custom prompt, since claude works best when prompted to output XML -- so you can prompt and then parse for XML

You can set `extract_prompt` in the constructor. This is the default prompt (there's two deepending on your options) https://github.com/run-llama/llama_index/blob/3c666ea32f98b26f0f96ebe82af67a3edf8d84ad/llama-index-core/llama_index/core/indices/property_graph/transformations/dynamic_llm.py#L235
